kratos/utilities/search_utilities.cpp

```diff
@@ -21,6 +21,72 @@
 namespace Kratos
 {
 
+bool SearchUtilities::PointIsInsideBoundingBox(
+    const BoundingBox<Point>& rBoundingBox,
+    const array_1d<double, 3>& rCoords
+    )
+{
+    // Get the bounding box points
+    const auto& r_max_point = rBoundingBox.GetMaxPoint();
+    const auto& r_min_point = rBoundingBox.GetMinPoint();
+
+    // The Bounding Box check
+    if (rCoords[0] < r_max_point[0] && rCoords[0] > r_min_point[0])           // check x-direction
+        if (rCoords[1] < r_max_point[1] && rCoords[1] > r_min_point[1])       // check y-direction
+            if (rCoords[2] < r_max_point[2] && rCoords[2] > r_min_point[2])   // check z-direction
+                return true;
+    return false;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+bool SearchUtilities::PointIsInsideBoundingBox(
+    const BoundingBoxType& rBoundingBox,
+    const array_1d<double, 3>& rCoords
+    )
+{
+    // The Bounding Box should have some tolerance already!
+    if (rCoords[0] < rBoundingBox[0] && rCoords[0] > rBoundingBox[1])           // check x-direction
+        if (rCoords[1] < rBoundingBox[2] && rCoords[1] > rBoundingBox[3])       // check y-direction
+            if (rCoords[2] < rBoundingBox[4] && rCoords[2] > rBoundingBox[5])   // check z-direction
+                return true;
+    return false;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+bool SearchUtilities::PointIsInsideBoundingBoxWithTolerance(
+    const BoundingBox<Point>& rBoundingBox,
+    const array_1d<double, 3>& rCoords,
+    const double Tolerance
+    )
+{
+    // Get the bounding box points
+    auto max_point = rBoundingBox.GetMaxPoint();
+    auto min_point = rBoundingBox.GetMinPoint();
+    
+    // Apply Tolerances (only in non zero BB cases)
+    const double epsilon = std::numeric_limits<double>::epsilon();
+    if (norm_2(max_point) > epsilon && norm_2(min_point) > epsilon) {
+        for (unsigned int i=0; i<3; ++i) {
+            max_point[i] += Tolerance;
+            min_point[i] -= Tolerance;
+        }
+    }
```

Comment:
This makes no sense to me. Why is this check any better than just checking the norm of the input point?

Furthermore, the docs read `This method checks if a point is inside any bounding box of the global bounding boxes considering a certain tolerance`, but the tolerance is applied **only if** the bounding box is triple-degenerate and exactly on the origin.

Reply:
Corrected the doc